### PR TITLE
1.21 Port

### PR DESCRIPTION
**Pull request by AzureAaron**
I've been keeping this updated myself for some time now, and I'm finally able to open a PR for it so heres's my 1.21 port!

I had to comment out the GitHub releases part because it broke the build script for me, I don't know that much about groovy let alone Kotlin so it's not something I can fix for you.

I've been using this for a while myself and we've also been distributing my versions in the [Skyblocker Modpack](https://modrinth.com/modpack/skyblocker-modpack/version/3.4.3) and nobody has reported anything to us about it being broken so I think everything should be ok.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,32 +1,28 @@
 [versions]
-loom = "1.3.+"
-loom_vineflower = "1.11.+"
+loom = "1.7.+"
 
 minotaur = "2.7.+"
```
Comment (Zalk0) [inline gradle/libs.versions.toml]:
```suggestion
minotaur = "2.8.+"
```
Why not bump ?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,32 +1,28 @@
...
 
-minecraft = "1.20.2"
-quilt_mappings = "2"
-fabric_loader = "0.14.22"
-fabric_api = "0.89.1+1.20.2"
-yacl = "3.2.1+1.20.2"
-mod_menu = "8.0.0"
-mixin_extras = "0.2.0-rc.4"
+minecraft = "1.21"
+quilt_mappings = "1"
```
Comment (Zalk0) [inline gradle/libs.versions.toml]:
```suggestion
quilt_mappings = "5"
```
Quilt mappings for MC 1.21 are at v5 (see https://quiltmc.org/en/usage/latest-versions/)

File: src/main/java/dev/isxander/adaptivetooltips/mixins/GameRendererMixin.java [hunk 3]
```diff
@@ -10,7 +10,7 @@
 @Mixin(GameRenderer.class)
 public class GameRendererMixin {
     @Inject(method = "render", at = @At("RETURN"))
-    private void checkTooltipRendered(float tickDelta, long startTime, boolean tick, CallbackInfo ci) {
+    private void checkTooltipRendered(CallbackInfo ci) {
```
Comment (Zalk0) [inline src/main/java/dev/isxander/adaptivetooltips/mixins/GameRendererMixin.java:13]:
```suggestion
    private void checkTooltipRendered(DeltaTracker tracker, boolean tick, CallbackInfo ci) {
```
Why remove parameters of the original method when there was only a change in parameters.
DeltaTracker would then need to be imported as such:
```java
import net.minecraft.client.DeltaTracker;
```

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -1,32 +1,28 @@
...
+fabric_loader = "0.15.11"
+fabric_api = "0.100.1+1.21"
```
Comment (Zalk0) [inline gradle/libs.versions.toml:12]:
```suggestion
fabric_api = "0.100.2+1.21"
```
I would have bumped one time as 0.100.1 was for MC 1.21-rc1 as we can see in the GitHub releases https://github.com/FabricMC/fabric/releases

**Review by Zalk0 — COMMENTED** [review]
I was working on the port too but you were faster, I'm just pointing some things out.
Also, in build.gradle.kts you can remove the repository https://jitpack.io as it was only used by MixinExtras which you removed because it is now bundled in fabric loader.
Good job!

File: src/main/java/dev/isxander/adaptivetooltips/mixins/GameRendererMixin.java (hunk 3 above)
Comment (isXander) [reply src/main/java/dev/isxander/adaptivetooltips/mixins/GameRendererMixin.java:13]: If the parameters aren't used they can be removed. This improves compatibility between versions.

Comment (Zalk0) [reply src/main/java/dev/isxander/adaptivetooltips/mixins/GameRendererMixin.java:13]: Ok, that's cool, I didn't know it was possible.

File: gradle/libs.versions.toml (hunk 1 above)
Comment (AzureAaron) [reply gradle/libs.versions.toml]: I was focused only on updating the mod for Minecraft I didn't pay much attention to anything else. I've updated it

File: gradle/libs.versions.toml (hunk 2 above)
Comment (AzureAaron) [reply gradle/libs.versions.toml]: Updated, when I first ported this on 1.21 release day Quilt didn't even have 1.21 mappings out so it used the rc lmao. On a side note Adaptive Tooltips should probably switch away from QM and use either Yarn or Mojmap given the development situation at Quilt.

File: gradle/libs.versions.toml (hunk 4 above)
Comment (AzureAaron) [reply gradle/libs.versions.toml:12]: Given that said Fabric API version is marked on MR as 1.21 compatible I don't think this is needed, the RCs are usually a near perfect match to the final release anyways.

**Comment by Zalk0** [comment]
@isXander any news on when this will be merged ?

**Comment by mucookul** [comment]
I went ahead and built it for the lazy people:

[AdaptiveTooltips-1.3.0-fabric-1.21.zip](https://github.com/user-attachments/files/16502527/AdaptiveTooltips-1.3.0-fabric-1.21.zip)

**Comment by isXander** [comment]
Thank you all for the help!

**Comment by mucookul** [comment]
@isXander When I try to set it to override Vanilla Tooltips, it crashes the game for me.

Edit: heres the log: https://mclo.gs/0bQHWoJ

Edit 2: heres a full log: https://mclo.gs/fx9OiEs, and it only seems to crash when trying to toggle the option, if you relaunch the game after the crash, it is enabled.

**Comment by isXander** [comment]
I found this today whilst sound some work to it. All good now

**Comment by Zalk0** [comment]
Can we expect a release soon ? (With the crash fixed maybe)

**Comment by notrohit1702** [comment]
Not to be a bother, but is there an estimated time till a Modrinth release?

**Comment by Zalk0** [comment]
@isXander do you plan on doing a release some day?
I see that you're working on other mods but I bet a lot of people would be delighted to have an official 1.21(.1) release.

**Comment by PseudoResonance** [comment]
Here's a build for 1.21.4 if anyone happens to come looking like me.

Requires Fabric Loader 0.16.10 and Fabric API 0.118.0+1.21.4 or higher.

[AdaptiveTooltips-1.3.0-fabric-1.21.4.jar.zip](https://github.com/user-attachments/files/19037801/AdaptiveTooltips-1.3.0-fabric-1.21.4.jar.zip)

**Pull request merged**